shared/media-accelerator/develop/ensure-channel-quality/project-implementation/android.mdx

```diff
@@ -0,0 +1,175 @@
+<PlatformWrapper platform="android">
+
+### Implement the user interface
+
+To enable app users to mute and unmute audio and video, add checkboxes to the user interface. To do this, open `/app/res/layout/activity_main.xml` and add the following lines before `</RelativeLayout>`:
+
+    ```xml
+    <CheckBox
+        android:id="@+id/MuteLocalAudio"
+        android:text="Mute local audio"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_below="@id/JoinButton"
+        android:layout_alignStart="@id/JoinButton" />
+
+    <CheckBox
+        android:id="@+id/MuteLocalVideo"
+        android:text="Mute local video"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_below="@id/MuteLocalAudio"
+        android:layout_alignStart="@id/JoinButton" />
+    ```
+
+### Handle the system logic
+
+1. **Import supporting libraries**
+
+    In `MainActivity.java`, add the following to the list of `import` statements:
+
+    ```java
+    import java.lang.Math;
+    import android.widget.CheckBox;
+    import android.widget.CompoundButton;
+    ```
+
+1. **Define variables to access checkboxes**
+
+    In `/app/java/com.example.<projectname>/MainActivity`, add the following declarations to the `MainActivity` class:
+
+    ```java
+    private CheckBox checkMuteLocalAudio;
+    private CheckBox checkMuteLocalVideo;
+    ```
+
+### Implement channel quality features
+
+1. **Configure the <Vpd k="SDK" /> log file**
+
+    To customize the location, and content of the log file, add the following code to `setupAgoraRtcService()` before `agoraRtcService.init(appId, agoraRtcEvents, options);`
```

Comment:
Ok, you will need to go back to the quickstart for this, but AgoraRtcService is the class, but we call it agoraEngine for all the other platforms, let's be consistent.

Reply:
Done